crates/wash-lib/src/common.rs

```diff
@@ -1,64 +1,125 @@
-use std::str::FromStr;
+use std::{
+    fmt::{Debug, Display},
+    str::FromStr,
+};
 
-use crate::id::ModuleId;
+use anyhow::Context;
+use wasmcloud_control_interface::HostInventory;
+
+use crate::id::{ModuleId, ServerId, ServiceId};
 
 const CLAIMS_CALL_ALIAS: &str = "call_alias";
 pub(crate) const CLAIMS_NAME: &str = "name";
 pub(crate) const CLAIMS_SUBJECT: &str = "sub";
 
 /// Converts error from Send + Sync error to standard anyhow error
 pub(crate) fn boxed_err_to_anyhow(e: Box<dyn ::std::error::Error + Send + Sync>) -> anyhow::Error {
-    anyhow::anyhow!(e.to_string())
+    anyhow::anyhow!(e)
 }
 
 #[derive(Debug, thiserror::Error)]
 pub enum FindIdError {
     /// No matches were found
-    #[error("No actor found with the search term")]
+    #[error("No matches found with the provided search term")]
     NoMatches,
-    /// Multiple matches were found. The vector contains the list of actors that matched
-    #[error("Multiple actors found with the search term: {0:?}")]
-    MultipleMatches(Vec<String>),
+    /// Multiple matches were found. The vector contains the list of actors or providers that
+    /// matched
+    #[error("Multiple matches found with the provided search term: {0:?}")]
+    MultipleMatches(Vec<Match>),
     #[error(transparent)]
     Error(#[from] anyhow::Error),
 }
 
-/// Given a string, attempts to resolve an actor ID. Returning the actor ID and an optional friendly name
+/// Represents a single match against a search term
+#[derive(Clone)]
+pub struct Match {
+    pub id: String,
+    pub friendly_name: Option<String>,
+}
+
+impl Display for Match {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        if let Some(friendly_name) = &self.friendly_name {
+            write!(f, "{} ({friendly_name})", self.id)
+        } else {
+            write!(f, "{}", self.id)
+        }
+    }
+}
+
+impl Debug for Match {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        Display::fmt(&self, f)
+    }
+}
```

Comment:
QUESTION: What do we get by using the display impl for debug vs the generated one? Just avoiding the default because it's sometimes hard to read?

Reply:
Good question. So I did this on purpose because I wanted it to print the same way when an error was printed whether or not it used `Display` or `Debug`. That way the error always prints nicely

Reply:
Ah OK got it, thanks for the explanation!